pycbc/detector.py

```diff
@@ -655,134 +655,810 @@ def overhead_antenna_pattern(right_ascension, declination, polarization):
     return f_plus, f_cross
 
 
-"""     LISA class      """
+"""     Space-based detector class      """
 
+from pycbc.coordinates.space import TIME_OFFSET_20_DEGREES
+from abc import ABC, abstractmethod
 
-class LISA(object):
-    """For LISA detector
+class AbsSpaceDet(ABC):
     """
-    def __init__(self):
-        None
+    Parent class to space-based detector classes.
+
+    Parameters
+    ----------
+    reference_time : float (optional)
```

Comment:
@acorreia61201 For a base class, you want to make sure it really is the minimal set of things that all implementations must have to work. I think there are few things here that don't necessarily qualify. It might be worth going through and pruning a bit here. 

A base class for example shouldn't have options just because a child class might have them. It's the child class's job to implement things that are specific to it.